tests/integration/integration_go_test.go

```diff
@@ -21,6 +21,26 @@ import (
 	ptesting "github.com/pulumi/pulumi/sdk/v3/go/common/testing"
 )
 
+// This checks that the buildTarget option for Pulumi Go programs does build a binary.
+func TestBuildTarget(t *testing.T) {
+	t.Parallel()
+
+	e := ptesting.NewEnvironment(t)
+	defer func() {
+		if !t.Failed() {
+			e.DeleteEnvironmentFallible()
+		}
+	}()
+	e.ImportDirectory(filepath.Join("go", "go-build-target"))
+
+	e.RunCommand("pulumi", "login", "--cloud-url", e.LocalURL())
```

Comment:
Curious why not use ProgramTest with some options like Quick: true or even further options? Do you need to drop to NewEnvironment? This will miss recent optimisations to ProgramTest like using filestate backend.

Reply:
I needed to get the working directory of the test to `stat` the buildTarget file is created and ProgramTest didn't have that facility